lib/src/facts/resolver.cc

```diff
@@ -24,6 +26,41 @@ namespace facter { namespace facts {
                 throw invalid_name_pattern_exception(ex.what());
             }
         }
+
+#ifdef LEATHERMAN_USE_LOCALES
+        // build Accept-Language list for HTTP-based resolvers
+        // check Leatherman then env vars for valid locale name
+        auto parse_lang = [](string str) {
+            static const boost::regex expr("([a-z]{2,}(_[a-z]{2,})?)(\\..+)?",
+                                           boost::regex::icase);
+            boost::smatch matches;
+            if (boost::regex_match(str, matches, expr)) {
+                str = matches[1];
+                std::transform(str.begin(), str.end(), str.begin(), ::tolower);
+                std::replace(str.begin(), str.end(), '_', '-');
+            } else {
+                str.clear();
+            }
+            return str;
+        };
+        string lang = parse_lang(leatherman::locale::get_locale().name());
```

Comment:
Are you asking about the string returned from
leatherman::locale::get_locale().name()? This string is not empty, but in
all the testing I've done, it returns "*", which indicates an unnamed
locale. It does this even when I explicitly request a valid named locale
(e.g. leatherman::locale::get_locale("fr_FR.UTF-8")). The locale does the
right thing (translation) but .name() is always "*".

So this code always falls back to determining the locale name from
environment variables. This is not ideal. It would be better to determine
locale name from the current Leatherman/Boost locale. Perhaps I should dig
deeper into boost::locale to understand why we get "*".

On Tue, Aug 2, 2016 at 9:11 AM, Michael Smith <notifications@github.com>
wrote:

> In lib/src/facts/resolver.cc
> <https://github.com/puppetlabs/facter/pull/1388#discussion_r73186540>:
>
> > +        // build Accept-Language list for HTTP-based resolvers
> > +        // check Leatherman then env vars for valid locale name
> > +        auto parse_lang = [](string str) {
> > +            static const boost::regex expr("([a-z]{2,}(_[a-z]{2,})?)(\\..+)?",
> > +                                           boost::regex::icase);
> > +            boost::smatch matches;
> > +            if (boost::regex_match(str, matches, expr)) {
> > +                str = matches[1];
> > +                std::transform(str.begin(), str.end(), str.begin(), ::tolower);
> > +                std::replace(str.begin(), str.end(), '_', '-');
> > +            } else {
> > +                str.clear();
> > +            }
> > +            return str;
> > +        };
> > +        string lang = parse_lang(leatherman::locale::get_locale().name());
>
> When is this ever empty?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/puppetlabs/facter/pull/1388/files/69cbaf149d40d2712ab9cd8d52f1679ded623d03#r73186540>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AMVyKu0pap7F9CtvyEnIWOMUzKLXzMbsks5qb2wigaJpZM4JaSQc>
> .
>



-- 
*Andy Henroid*
Senior Software Engineer
andy.henroid@puppet.com